lib/Service/CustomerService.php

```diff
@@ -0,0 +1,267 @@
+    {
+        return $this->deleteNestedObject(self::TAX_IDS, $customerId, $id, $params, $opts);
+    }
+
+    /**
+     * Retrieve a customer.
+     *
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Customer
+     */
+    public function retrieve($id, $params = [], $opts = [])
+    {
+        return $this->retrieveObject($id, $params, $opts);
+    }
+
+    /**
+     * Retrieve a balance transaction.
+     *
+     * @param string $customerId
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\CustomerBalanceTransaction
+     */
+    public function retrieveBalanceTransaction($customerId, $id, $params = [], $opts = [])
+    {
+        return $this->retrieveNestedObject(self::BALANCE_TRANSACTIONS, $customerId, $id, $params, $opts);
+    }
+
+    /**
+     * Retrieve a source.
+     *
+     * @param string $customerId
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Account|\Stripe\AlipayAccount|\Stripe\BankAccount|\Stripe\BitcoinReceiver|\Stripe\Card|\Stripe\Source
+     */
+    public function retrieveSource($customerId, $id, $params = [], $opts = [])
+    {
+        return $this->retrieveNestedObject(self::SOURCES, $customerId, $id, $params, $opts);
+    }
+
+    /**
+     * Retrieve a tax ID.
+     *
+     * @param string $customerId
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\TaxId
+     */
+    public function retrieveTaxId($customerId, $id, $params = [], $opts = [])
+    {
+        return $this->retrieveNestedObject(self::TAX_IDS, $customerId, $id, $params, $opts);
+    }
+
+    /**
+     * Update a customer.
+     *
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Customer
+     */
+    public function update($id, $params = [], $opts = [])
+    {
+        return $this->updateObject($id, $params, $opts);
+    }
+
+    /**
+     * Update a balance transaction.
+     *
+     * @param string $customerId
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\CustomerBalanceTransaction
+     */
+    public function updateBalanceTransaction($customerId, $id, $params = [], $opts = [])
+    {
+        return $this->updateNestedObject(self::BALANCE_TRANSACTIONS, $customerId, $id, $params, $opts);
+    }
+
+    /**
+     * Update a source.
+     *
+     * @param string $customerId
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Account|\Stripe\AlipayAccount|\Stripe\BankAccount|\Stripe\BitcoinReceiver|\Stripe\Card|\Stripe\Source
+     */
+    public function updateSource($customerId, $id, $params = [], $opts = [])
+    {
+        return $this->updateNestedObject(self::SOURCES, $customerId, $id, $params, $opts);
+    }
+}
```

Comment:
We should talk about how we'll do custom methods on nested resources like verifying ACH.

Reply:
I added a `verifySource` method.